src/app/tests/TestReadInteraction.cpp

```diff
@@ -986,6 +999,88 @@ void TestReadInteraction::TestReadRoundtripWithSameDifferentPathsDataVersionFilt
+    err           = engine->Init(&ctx.GetExchangeManager());
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    EventManagement::GetInstance().SetBypassACL(chip::app::EventManagement::BypassACL::kAlwaysFail);
+
+    // When read events with concrete paths without enough privilege, we will return EventStatusIB
+    {
+        chip::app::EventPathParams eventPathParams[1];
+        eventPathParams[0].mEndpointId = kTestEndpointId;
+        eventPathParams[0].mClusterId  = kTestClusterId;
+        eventPathParams[0].mEventId    = kTestEventIdDebug;
+
+        ReadPrepareParams readPrepareParams(ctx.GetSessionBobToAlice());
+        readPrepareParams.mpEventPathParamsList    = eventPathParams;
+        readPrepareParams.mEventPathParamsListSize = 1;
+        readPrepareParams.mEventNumber             = 1;
+
+        MockInteractionModelApp delegate;
+        NL_TEST_ASSERT(apSuite, !delegate.mGotEventResponse);
+
+        app::ReadClient readClient(chip::app::InteractionModelEngine::GetInstance(), &ctx.GetExchangeManager(), delegate,
+                                   chip::app::ReadClient::InteractionType::Read);
+
+        err = readClient.SendRequest(readPrepareParams);
+        NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+        InteractionModelEngine::GetInstance()->GetReportingEngine().Run();
+        NL_TEST_ASSERT(apSuite, delegate.mGotEventResponse);
+        NL_TEST_ASSERT(apSuite, delegate.mNumReadEventFailureStatusReceived > 0);
+        NL_TEST_ASSERT(apSuite, delegate.mLastStatusReceived.mStatus == Protocols::InteractionModel::Status::UnsupportedAccess);
+        NL_TEST_ASSERT(apSuite, !delegate.mReadError);
+    }
+
+    GenerateEvents(apSuite, apContext);
+
+    EventManagement::GetInstance().SetBypassACL(chip::app::EventManagement::BypassACL::kAlwaysFail);
+    // When read events with withcard paths without enough privilege for reading one or more event, we will not exclude the events
+    // when generating the report.
+    {
+        chip::app::EventPathParams eventPathParams[1];
+        eventPathParams[0].mEndpointId = kTestEndpointId;
+        eventPathParams[0].mClusterId  = kTestClusterId;
+
+        ReadPrepareParams readPrepareParams(ctx.GetSessionBobToAlice());
+        readPrepareParams.mpEventPathParamsList    = eventPathParams;
+        readPrepareParams.mEventPathParamsListSize = 1;
+        readPrepareParams.mEventNumber             = 1;
+
+        MockInteractionModelApp delegate;
+        NL_TEST_ASSERT(apSuite, !delegate.mGotEventResponse);
+
+        app::ReadClient readClient(chip::app::InteractionModelEngine::GetInstance(), &ctx.GetExchangeManager(), delegate,
+                                   chip::app::ReadClient::InteractionType::Read);
+
+        err = readClient.SendRequest(readPrepareParams);
+        NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+        InteractionModelEngine::GetInstance()->GetReportingEngine().Run();
```

Comment:
Again, please no manual Run calls.

Reply:
manual run call has been removed in tests in master branch, consider to rebase it. @erjiaqing